File: BaseProject.UI/src/app/user/shared/user.service.ts
import {inject, Injectable} from '@angular/core';
import {User, UserGeneratedService} from '@api/baseproject';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  private readonly userService = inject(UserGeneratedService)

  createUser(user : User) {
    return this.userService.createUser(user);
  }

  getAllUsers() {
    return this.userService.getAllUsers();
  }
}

File: BaseProject.Infrastructure/Service/UserService.cs
using BaseProject.Domain.Contracts;
using BaseProject.Domain.Model;
using BaseProject.Infrastructure.Database;
using Microsoft.EntityFrameworkCore;

namespace BaseProject.Infrastructure.Service;

public class UserService : IUserService
{
	private readonly BaseProjectDBContext _context;

	public UserService(BaseProjectDBContext context)
	{
		_context = context;
	}

	public async Task<List<User>> GetAllUsers()
	{
		return await _context.Users.ToListAsync();
	}
	
	public async Task CreateUser(User entity)
	{
		await _context.Users.AddAsync(entity);
	}
}